client/Hooks/types.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { FunctionComponent } from 'react';
+
+export type PageProperties = {
```

Comment:
Good point. I was curious as to where a good place to keep these types would be. I quite like them being outside of the source file but I can see that getting massive if it's types for all hooks. Do you think it would be worth having a `Hooks/types` folder that has the types for each file that can be accessed through a barrel file? That way we keep a neat folder structure and can keep the types separately

Reply:
And yeah, forgot to as JSDoc

Reply:
> Good point. I was curious as to where a good place to keep these types would be. I quite like them being outside of the source file but I can see that getting massive if it's types for all hooks. Do you think it would be worth having a `Hooks/types` folder that has the types for each file that can be accessed through a barrel file? That way we keep a neat folder structure and can keep the types separately
`
If a type is used in multiple files, then it being in a seperate file (`*.assets.ts`) - eg https://github.com/matthew-chirgwin/strimzi-ui/blob/13-integrate/client/Contexts/ConfigFeatureFlag/ConfigFeatureFlag.assets.ts  is what I would suggest for now. The users of types will be developers, who likely either want guidance on what X is, or want to type X as something. Thus, I think the types being a constant/exported item from the thing that defines them makes sense. I am not sure all the types being in one file or sub directory would make sense, as it breaks them away from the thing(s) which use them.

Reply:
@nictownsend Yep! Already making that approach now